### ISSUE TITLE
fastlane: command not found

### ISSUE THREAD

**Issue by HuangWei201703** [fastlane/fastlane#9409]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

I am sorry, I am a chinese, I can't speak English, so, I was using Jenkins to upload IPA times wrong : 

```
Archiving dSYM
[H20170531_OCUnit] $ /bin/sh -xe /Users/Shared/Jenkins/tmp/hudson7322474858661619663.sh
+ IPANAME=$'\342\200\234H20170531_OCUnit\342\200\235'
+ fastlane gym --export_method development --output_name $'\342\200\234H20170531_OCUnit\342\200\235'
/Users/Shared/Jenkins/tmp/hudson7322474858661619663.sh: line 3: fastlane: command not found
```

[INSERT OUTPUT HERE]

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.5                                     |
| Ruby                        | 2.4.1                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.9.2                           |
| Installation Source         | /usr/local/bin/fastlane                     |
| Host                        | Mac OS X 10.12.5 (16F73)                    |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.4.1_1/lib          |
| OpenSSL Version             | OpenSSL 1.0.2k  26 Jan 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.3                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | zh_CN.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.37.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.2.1        |
| terminal-table            | 1.8.0        |
| plist                     | 3.3.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.4        |
| excon                     | 0.56.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.28       |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.1        |
| bundler                   | 1.15.1       |
| faraday_middleware        | 0.11.0.1     |
| json                      | 2.1.0        |
| io-console                | 0.4.6        |
| openssl                   | 2.0.3        |
</details>

*generated on:* **2017-06-08**
</details>

### Environment
 
Jenkins ver. 2.46.3

**Comment by getaaron** [comment]
This is probably an issue with your Jenkins configuration. You may need to set the `PATH` environment variable. On my Jenkins configuration `PATH` is set to:

```
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

This works for me because `fastlane` is located in `/usr/local/bin/fastlane`.

**Comment by HuangWei201703** [comment]
Thank you very much!

This is my Jenkins configuration:

![wx20170609-101526](https://user-images.githubusercontent.com/26423316/26958487-9641d9f2-4cfd-11e7-8923-5d5a92d8e0f5.png)

![wx20170609-101502](https://user-images.githubusercontent.com/26423316/26958494-a37a59e6-4cfd-11e7-92c2-759e8dfcd1d3.png)

Use Jenkins in this configuration execute shell error:

```
fastlane: command not found
```

my shell:
```
IPANAME=“SWARM”
fastlane gym --export_method ad-hoc --output_name ${IPANAME}
curl -F "file=@${IPANAME}.ipa" -F "uKey=c14292cxxxxx1013b08a" -F "_api_key=0c970f3f5a3dxxxxx4096acb1" https://qiniu-storage.pgyer.com/apiv1/app/upload
```

**Comment by getaaron** [comment]
Are you sure fastlane is installed on your build agent? How do you know?

**Comment by HuangWei201703** [comment]
```
hwdeMacBook-Pro:H20170531_OCUnit HW$ which fastlane
/Users/HW/.fastlane/bin/fastlane
```

```
hwdeMacBook-Pro:H20170531_OCUnit HW$ fastlane gym --verbose
INFO [2017-06-12 14:40:01.40]: $ xcodebuild -list -workspace ./H20170531_OCUnit.xcworkspace
2017-06-12 14:40:01.913 xcodebuild[2230:111789] CoreSimulator detected Xcode.app relocation or CoreSimulatorService version change.  Framework path (/Applications/Xcode.app/Contents/Developer/Library/PrivateFrameworks/CoreSimulator.framework) and version (375.21) does not match existing job path (/Library/Developer/PrivateFrameworks/CoreSimulator.framework/Versions/A/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc) and version (459.9).  Attempting to remove the stale service in order to add the expected version.
Select Scheme:
1. H20170531_OCUnit
2. OCHamcrest
3. OCMockito
4. Pods-H20170531_OCUnit
5. Pods-H20170531_OCUnitTests
?  1
INFO [2017-06-12 14:40:04.18]: $ xcodebuild -showBuildSettings -workspace ./H20170531_OCUnit.xcworkspace -scheme H20170531_OCUnit

+----------------------+--------------------------------------+
|                   Summary for gym 2.28.3                    |
+----------------------+--------------------------------------+
| workspace            | ./H20170531_OCUnit.xcworkspace       |
| scheme               | H20170531_OCUnit                     |
| destination          | generic/platform=iOS                 |
| output_name          | H20170531_OCUnit                     |
| build_path           | /Users/HW/Library/Developer/Xcode/A  |
|                      | rchives/2017-06-12                   |
| clean                | false                                |
| output_directory     | .                                    |
| silent               | false                                |
| skip_package_ipa     | false                                |
| use_legacy_build_api | false                                |
| buildlog_path        | ~/Library/Logs/gym                   |
| xcode_path           | /Applications/Xcode.app              |
+----------------------+--------------------------------------+

+--------------------------------------+----------------------+
|                   Generated Build Command                   |
+--------------------------------------+----------------------+
| Option                               | Value                |
+--------------------------------------+----------------------+
| set -o pipefail &&                   |                      |
| xcodebuild                           |                      |
| -workspace                           |                      |
| ./H20170531_OCUnit.xcworkspace       |                      |
| -scheme H20170531_OCUnit             |                      |
| -destination                         | generic/platform=iOS |
| -archivePath                         |                      |
| /Users/HW/Library/Developer/Xcode/A  |                      |
| rchives/2017-06-12/H20170531_OCUnit  |                      |
| \ 2017-06-12\ 14.40.05.xcarchive     |                      |
| archive                              |                      |
| | tee                                |                      |
| /Users/HW/Library/Logs/gym/H2017053  |                      |
| 1_OCUnit-H20170531_OCUnit.log        |                      |
| | xcpretty                           |                      |
+--------------------------------------+----------------------+
INFO [2017-06-12 14:40:05.29]: $ set -o pipefail && xcodebuild -workspace ./H20170531_OCUnit.xcworkspace -scheme H20170531_OCUnit -destination 'generic/platform=iOS' -archivePath /Users/HW/Library/Developer/Xcode/Archives/2017-06-12/H20170531_OCUnit\ 2017-06-12\ 14.40.05.xcarchive archive | tee /Users/HW/Library/Logs/gym/H20170531_OCUnit-H20170531_OCUnit.log | xcpretty
INFO [2017-06-12 14:40:06.22]: ▸ Building Pods/Pods-H20170531_OCUnit [Release]
INFO [2017-06-12 14:40:06.22]: ▸ Check Dependencies
INFO [2017-06-12 14:40:06.29]: ▸ Compiling Pods-H20170531_OCUnit-dummy.m
INFO [2017-06-12 14:40:06.47]: ▸ Compiling Pods-H20170531_OCUnit-dummy.m
INFO [2017-06-12 14:40:06.47]: ▸ Building library libPods-H20170531_OCUnit.a
INFO [2017-06-12 14:40:06.48]: ▸ Building library libPods-H20170531_OCUnit.a
INFO [2017-06-12 14:40:06.50]: ▸ Building H20170531_OCUnit/H20170531_OCUnit [Release]
INFO [2017-06-12 14:40:06.50]: ▸ Check Dependencies
INFO [2017-06-12 14:40:06.52]: ▸ Running script '[CP] Check Pods Manifest.lock'
INFO [2017-06-12 14:40:06.54]: ▸ Compiling Counter.m
INFO [2017-06-12 14:40:06.74]: ▸ Compiling CounterViewController.m
INFO [2017-06-12 14:40:06.79]: ▸ Compiling AppDelegate.m
INFO [2017-06-12 14:40:06.79]: ▸ Compiling main.m
INFO [2017-06-12 14:40:06.79]: ▸ Compiling Counter.m
INFO [2017-06-12 14:40:06.91]: ▸ Compiling CounterViewController.m
INFO [2017-06-12 14:40:06.93]: ▸ Compiling AppDelegate.m
INFO [2017-06-12 14:40:06.94]: ▸ Compiling main.m
INFO [2017-06-12 14:40:06.94]: ▸ Linking H20170531_OCUnit
INFO [2017-06-12 14:40:07.02]: ▸ ⚠️  ld: directory not found for option '-L/Users/HW/Documents/1.about iOS/4.iOSStudy/H20170531_OCUnit/build/Debug-iphoneos'
INFO [2017-06-12 14:40:07.02]: ▸ Linking H20170531_OCUnit
INFO [2017-06-12 14:40:07.02]: ▸ ⚠️  ld: directory not found for option '-L/Users/HW/Documents/1.about iOS/4.iOSStudy/H20170531_OCUnit/build/Debug-iphoneos'
INFO [2017-06-12 14:40:07.04]: ▸ Compiling LaunchScreen.storyboard
INFO [2017-06-12 14:40:08.66]: ▸ Compiling Main.storyboard
INFO [2017-06-12 14:40:08.88]: ▸ Processing Info.plist
INFO [2017-06-12 14:40:08.89]: ▸ Generating 'H20170531_OCUnit.app.dSYM'
INFO [2017-06-12 14:40:09.05]: ▸ Running script '[CP] Embed Pods Frameworks'
INFO [2017-06-12 14:40:09.08]: ▸ Running script '[CP] Copy Pods Resources'
INFO [2017-06-12 14:40:10.11]: ▸ Touching H20170531_OCUnit.app
INFO [2017-06-12 14:40:10.12]: ▸ Signing /Users/HW/Library/Developer/Xcode/DerivedData/H20170531_OCUnit-egngmwhayvtyqgbpjpemjqwiiccm/Build/Intermediates/ArchiveIntermediates/H20170531_OCUnit/InstallationBuildProductsLocation/Applications/H20170531_OCUnit.app
INFO [2017-06-12 14:40:10.28]: ▸ Touching H20170531_OCUnit.app.dSYM
INFO [2017-06-12 14:40:10.39]: ▸ Archive Succeeded
DEBUG [2017-06-12 14:40:10.74]: Stored the archive in: /Users/HW/Library/Developer/Xcode/Archives/2017-06-12/H20170531_OCUnit 2017-06-12 14.40.05.xcarchive
WARN [2017-06-12 14:40:10.74]: Generated plist file with the following values:
INFO [2017-06-12 14:40:10.74]: ▸ -----------------------------------------
INFO [2017-06-12 14:40:10.74]: ▸ {
INFO [2017-06-12 14:40:10.74]: ▸   "method": "app-store",
INFO [2017-06-12 14:40:10.74]: ▸   "uploadSymbols": true,
INFO [2017-06-12 14:40:10.74]: ▸   "uploadBitcode": false
INFO [2017-06-12 14:40:10.74]: ▸ }
INFO [2017-06-12 14:40:10.75]: ▸ -----------------------------------------
INFO [2017-06-12 14:40:10.75]: This results in the following plist file:
INFO [2017-06-12 14:40:10.75]: ▸ -----------------------------------------
INFO [2017-06-12 14:40:10.75]: ▸ <?xml version="1.0" encoding="UTF-8"?>
INFO [2017-06-12 14:40:10.75]: ▸ <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
INFO [2017-06-12 14:40:10.75]: ▸ <plist version="1.0">
INFO [2017-06-12 14:40:10.75]: ▸ <dict>
INFO [2017-06-12 14:40:10.75]: ▸ 	<key>method</key>
INFO [2017-06-12 14:40:10.75]: ▸ 	<string>app-store</string>
INFO [2017-06-12 14:40:10.75]: ▸ 	<key>uploadBitcode</key>
INFO [2017-06-12 14:40:10.75]: ▸ 	<false/>
INFO [2017-06-12 14:40:10.75]: ▸ 	<key>uploadSymbols</key>
INFO [2017-06-12 14:40:10.75]: ▸ 	<true/>
INFO [2017-06-12 14:40:10.75]: ▸ </dict>
INFO [2017-06-12 14:40:10.75]: ▸ </plist>
INFO [2017-06-12 14:40:10.75]: ▸ -----------------------------------------
+--------------------------------------+--------------------------------------+
|                          Generated Package Command                          |
+--------------------------------------+--------------------------------------+
| Option                               | Value                                |
+--------------------------------------+--------------------------------------+
| /usr/bin/xcrun                       |                                      |
| /Users/HW/.fastlane/bin/bundle/lib/  |                                      |
| ruby/gems/2.2.0/gems/fastlane-2.28.  |                                      |
| 3/gym/lib/assets/wrap_xcodebuild/xc  |                                      |
| build-safe.sh -exportArchive         |                                      |
| -exportOptionsPlist                  | /var/folders/9d/jrxjl4_14tj1w33z34j  |
|                                      | krxjm0000gn/T/gym_config20170612-22  |
|                                      | 16-17ps81z.plist                     |
| -archivePath                         |                                      |
| /Users/HW/Library/Developer/Xcode/A  |                                      |
| rchives/2017-06-12/H20170531_OCUnit  |                                      |
| \ 2017-06-12\ 14.40.05.xcarchive     |                                      |
| -exportPath                          | /var/folders/9d/jrxjl4_14tj1w33z34j  |
|                                      | krxjm0000gn/T/gym_output20170612-22  |
|                                      | 16-1sb1l3k                           |
+--------------------------------------+--------------------------------------+
INFO [2017-06-12 14:40:10.76]: $ /usr/bin/xcrun /Users/HW/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.28.3/gym/lib/assets/wrap_xcodebuild/xcbuild-safe.sh -exportArchive -exportOptionsPlist '/var/folders/9d/jrxjl4_14tj1w33z34jkrxjm0000gn/T/gym_config20170612-2216-17ps81z.plist' -archivePath /Users/HW/Library/Developer/Xcode/Archives/2017-06-12/H20170531_OCUnit\ 2017-06-12\ 14.40.05.xcarchive -exportPath '/var/folders/9d/jrxjl4_14tj1w33z34jkrxjm0000gn/T/gym_output20170612-2216-1sb1l3k'
INFO [2017-06-12 14:40:10.93]: ▸ RVM detected, forcing to use system ruby
INFO [2017-06-12 14:40:10.95]: ▸ Warning! PATH is not properly set up, '/Users/HW/.rvm/gems/ruby-2.2.5/bin' is not at first place.
INFO [2017-06-12 14:40:10.95]: ▸ Usually this is caused by shell initialization files. Search for 'PATH=...' entries.
INFO [2017-06-12 14:40:10.95]: ▸ You can also re-add RVM to your profile by running: 'rvm get stable --auto-dotfiles'.
INFO [2017-06-12 14:40:10.95]: ▸ To fix it temporarily in this shell session run: 'rvm use ruby-2.2.5'.
INFO [2017-06-12 14:40:10.95]: ▸ To ignore this error add rvm_silence_path_mismatch_check_flag=1 to your ~/.rvmrc file.
INFO [2017-06-12 14:40:11.16]: ▸ Now using system ruby.
INFO [2017-06-12 14:40:11.19]: ▸ + xcodebuild -exportArchive -exportOptionsPlist /var/folders/9d/jrxjl4_14tj1w33z34jkrxjm0000gn/T/gym_config20170612-2216-17ps81z.plist -archivePath '/Users/HW/Library/Developer/Xcode/Archives/2017-06-12/H20170531_OCUnit 2017-06-12 14.40.05.xcarchive' -exportPath /var/folders/9d/jrxjl4_14tj1w33z34jkrxjm0000gn/T/gym_output20170612-2216-1sb1l3k
INFO [2017-06-12 14:40:11.61]: ▸ 2017-06-12 14:40:11.610 xcodebuild[2499:113479] [MT] IDEDistribution: -[IDEDistributionLogging _createLoggingBundleAtPath:]: Created bundle at path '/var/folders/9d/jrxjl4_14tj1w33z34jkrxjm0000gn/T/H20170531_OCUnit_2017-06-12_14-40-11.609.xcdistributionlogs'.
INFO [2017-06-12 14:40:11.69]: ▸ 1.2.840.113635.100.1.61
INFO [2017-06-12 14:40:12.16]: ▸ Exported H20170531_OCUnit 2017-06-12 14.40.05.xcarchive to: /var/folders/9d/jrxjl4_14tj1w33z34jkrxjm0000gn/T/gym_output20170612-2216-1sb1l3k
INFO [2017-06-12 14:40:12.16]: ▸ ** EXPORT SUCCEEDED **
INFO [2017-06-12 14:40:12.17]: Compressing 1 dSYM(s)
INFO [2017-06-12 14:40:12.17]: $ cd '/Users/HW/Library/Developer/Xcode/Archives/2017-06-12/H20170531_OCUnit 2017-06-12 14.40.05.xcarchive/dSYMs' && zip -r '/Users/HW/Documents/1.about iOS/4.iOSStudy/H20170531_OCUnit/H20170531_OCUnit.app.dSYM.zip' *.dSYM
INFO [2017-06-12 14:40:12.29]: ▸   adding: H20170531_OCUnit.app.dSYM/ (stored 0%)
INFO [2017-06-12 14:40:12.29]: ▸   adding: H20170531_OCUnit.app.dSYM/Contents/ (stored 0%)
INFO [2017-06-12 14:40:12.29]: ▸   adding: H20170531_OCUnit.app.dSYM/Contents/Info.plist (deflated 52%)
INFO [2017-06-12 14:40:12.29]: ▸   adding: H20170531_OCUnit.app.dSYM/Contents/Resources/ (stored 0%)
INFO [2017-06-12 14:40:12.29]: ▸   adding: H20170531_OCUnit.app.dSYM/Contents/Resources/DWARF/ (stored 0%)
INFO [2017-06-12 14:40:12.29]: ▸   adding: H20170531_OCUnit.app.dSYM/Contents/Resources/DWARF/H20170531_OCUnit (deflated 62%)

INFO [2017-06-12 14:40:12.29]: Successfully exported and compressed dSYM file
INFO [2017-06-12 14:40:12.29]: Successfully exported and signed the ipa file:
INFO [2017-06-12 14:40:12.29]: /Users/HW/Documents/1.about iOS/4.iOSStudy/H20170531_OCUnit/H20170531_OCUnit.ipa

#######################################################################
# fastlane 2.37.0 is available. You are on 2.28.3.
# You should use the latest version.
# Please update using `fastlane update_fastlane`.
#######################################################################

2.37.0 Update for new iTunes Connect
* Add full support for new promotional text support in iTC (#9370) via Felix Krause
* Fix spaceship app_version to work with updated iTC (#9368) via Felix Krause
* Skip the upload success message when skipping the upload (#9352) via David Ohayon
* Allow for case insensitive screenshot directory names (#9349) via David Ohayon
* Remove invalid gym xcpretty options verify blocks (#9344) via Björn Dahlgren
* Add raise_shell_error option for sh action (#9313) via Kohki Miki
* refactor for newest version of rating endpoint (#9334) via Helmut Januschka
* Add a retry option on getting builds for a train (#9331) via David Ohayon
* Update GitHub URLs in commands_generator across tools (#9357) via Felix Krause

2.36.0 Improvements
* Allow for 1% rollouts in supply command (#9296) via Pan Thomakos
* Explicitly crash when the plist cant be parsed for provisioning profile data (#9319) via David Ohayon
* List produce enable_services options in config item (deprecating enabled_features) (#9223) via Josh Holtz
* Fix encoding error in update_url_schemes fastlane action (#9302) via Jochen Pfeiffer
* Fix spaceship crash in some edge cases (#9308) via Murat Çorlu
* Fix scan’s custom report name not working with multiple output types (#9303) via Lyndsey Ferguson
* Appaloosa action proxy support (#9310) via Robin
* Improve error message for code signing errors (#9291) via Joshua Liebowitz
* Fixed scan results for schemes that contain spaces (#9322) via NachoSoto
* [pilot] Show number of testers on top of table (#9323) via Felix Krause
* Improve error reporting (#9277) (#9295) via David Ohayon

2.35.1 Scan hotfix
* [scan] Use pre-2.34.0 strategy if temp junit output is missing (#9276) via Keaton Greve
* Fix puts crash on linux (#9279) via David Ohayon
* Warn user when before_all, after_all or error block overwrite each other (#9284) via Felix Krause
* Add helpful output when running `pilot` without specifying an action (#9283) via Felix Krause
* Update links to new docs page in crashlytics beta class (#9264) via Felix Krause

To see all new releases, open https://github.com/fastlane/fastlane/releases

Please update using `fastlane update_fastlane`
```
